### PR TITLE
Do not send back the ranks in communicateResultsBack

### PR DESCRIPTION
**Pull request by aprokop**
We never use them.

### CODE REVIEW

File: src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp [hunk 1]
```diff
@@ -606,11 +605,9 @@ void DistributedSearchTreeImpl<DeviceType>::communicateResultsBack(
       Kokkos::ViewAllocateWithoutInitializing(ids.label()), n_imports);
 
   sendAcrossNetwork(distributor, export_indices, import_indices);
-  sendAcrossNetwork(distributor, export_ranks, import_ranks);
```
Comment (masterleinad) [inline src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp]: But then there is also no point in declaring `export_ranks` and `import_ranks` at all.

File: src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp [hunk 2]
```diff
@@ -98,7 +98,7 @@ struct DistributedSearchTreeImpl
   static void communicateResultsBack(
       MPI_Comm comm, Kokkos::View<int *, DeviceType> &indices,
       Kokkos::View<int *, DeviceType> offset,
-      Kokkos::View<int *, DeviceType> &ranks,
```
Comment (masterleinad) [inline src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp]: What's the motivation for this change?

Comment (aprokop) [reply src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp]: By `&` we usually indicate that the view is getting modified inside. As this is not the case here, it seems reasonable to remove it.

File: src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp (hunk 1 above)
Comment (aprokop) [reply src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp]: Sorry, you are right. Was just winging it. Will force push with them removed.

File: src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp (hunk 2 above)
Comment (masterleinad) [reply src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp]: Hmm... I think putting `const` somewhere is much clearer but that discussion doesn't belong here.

**Comment by aprokop** [comment]
I just realized that this change only works all callbacks are implemented in distributed. It cannot pass the tests right now.

**Comment by aprokop** [comment]
@masterleinad Can you please block this until callbacks are merged?

**Comment by aprokop** [comment]
Closing for now. Will make it a part of the nearest callbacks PR later.

**Comment by masterleinad** [comment]
> Closing for now. Will make it a part of the nearest callbacks PR later.

Your decision. Small pull requests are most of the time easier to review, though.
